core/src/replay_stage.rs

```diff
@@ -138,7 +139,16 @@ impl ReplayStage {
                     let votable = Self::generate_votable_banks(&bank_forks, &tower, &mut progress);
 
                     if let Some((_, bank)) = votable.last() {
-                        subscriptions.notify_subscribers(bank.slot(), &bank_forks);
+                        if let Some(new_leader) =
```

Comment:
I chose this place to acknowledge the new leader so that if a leader is not active for their slot, I would not record them being an active leader.

Reply:
this spot is where that bank is being marked as *done*

Reply:
if we're trying to do a linear history of leaders, the vote stage is the closest thing to "right", but it won't be live, it'll be one slot late (which is probably fine)

if we're trying to say "now following XXX", we should just print the leaders of all the forks we're currently following, including our own when we're leader.

Reply:
My intention was to do do the former. Show the linear history of leaders but sacrifice some liveliness by waiting until the end of the slot to announce that the leader changed. As you said, I was thinking it would be fine to have that delay!